### PR TITLE
Rarity query

### PR DESCRIPTION
**Pull request by lxgr-linux**

### CODE REVIEW

File: docs/static/openapi.yml [hunk 1]
```diff
@@ -33242,34 +33242,51 @@ paths:
...
-            - bannedVerySoon
-            - none
-            - playable
-            - unplayable
         - name: owner
           in: query
           required: false
           type: string
-        - name: cardType
+        - name: statuses
```
Comment (patrickwieth) [inline docs/static/openapi.yml:33249]: why plural? a card can only have 1 status right?

File: x/cardchain/client/cli/query_q_cards.go [hunk 2]
```diff
@@ -14,27 +14,83 @@ var _ = strconv.Itoa(0)
 
 func CmdQCards() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "q-cards [owner] [status] [card-type] [classes] [sort-by] [name-contains] [keywords-contains] [notes-contains] [startercards-only] [balance-achors-only]",
+		Use:   "q-cards [owner] [statuses] [card-types] [classes] [rarities] [sort-by] [name-contains] [keywords-contains] [notes-contains] [startercards-only] [balance-achors-only]",
```
Comment (patrickwieth) [inline x/cardchain/client/cli/query_q_cards.go:17]: same question for card-type? A card can only have 1 type, so why is it plural?

File: x/cardchain/keeper/grpc_query_q_cards.go [hunk 3]
```diff
@@ -37,28 +37,7 @@ func (k Keeper) QCards(goCtx context.Context, req *types.QueryQCardsRequest) (*t
...
-			types.Status_prototype,
-			types.Status_suspended,
-			types.Status_banned,
-			types.Status_bannedSoon,
-			types.Status_bannedVerySoon,
-		}
-	default:
-		states = []types.Status{types.Status(types.Status_value[req.Status.String()])}
-	}
+	fmt.Printf("---> Hier: %#v\n", req)
```
Comment (patrickwieth) [inline x/cardchain/keeper/grpc_query_q_cards.go:40]: debug print important for production?

File: x/cardchain/keeper/grpc_query_q_cards.go [hunk 4]
```diff
@@ -73,35 +52,23 @@ func (k Keeper) QCards(goCtx context.Context, req *types.QueryQCardsRequest) (*t
...
-			if bool(cardobjClass.Nature) != strings.Contains(req.Classes, "Nature") {
-				return false
-			}
-			if bool(cardobjClass.Technology) != strings.Contains(req.Classes, "Technology") {
-				return false
-			}
-			if bool(cardobjClass.Culture) != strings.Contains(req.Classes, "Culture") {
-				return false
-			}
+		fmt.Printf("%v, %v\n", cardobjClass, req.Classes)
```
Comment (patrickwieth) [inline x/cardchain/keeper/grpc_query_q_cards.go:55]: debug print?

**Review by patrickwieth — COMMENTED** [review]
as a side note:
I have seen sometimes our blockchain logs print arrays with numbers that are hard to make any use of :D
looks like [123] [467] ...

File: docs/static/openapi.yml (hunk 1 above)
Comment (lxgr-linux) [reply docs/static/openapi.yml:33249]: yes, but dont u query a list of cards?

File: x/cardchain/keeper/grpc_query_q_cards.go (hunk 3 above)
Comment (lxgr-linux) [reply x/cardchain/keeper/grpc_query_q_cards.go:40]: huuch

File: docs/static/openapi.yml (hunk 1 above)
Comment (patrickwieth) [reply docs/static/openapi.yml:33249]: Eh yes, but you can only specify 1 status? So what you say is like "Give me all card**s** with status "permanent". 
Example from real life:
You want to buy 4 apple**s**, the apples are plural, then the nice lady in the marketplace asks you "Which sort of apples do you want?", then you yell "BOSKOP". The lady does not ask "Which sort**s** of apples do you want?"

**Review by patrickwieth — APPROVED** [review]

**Pull request merged**